### ISSUE TITLE
Using setServerLogCb Causes Python Process To Stay Open Forever

### ISSUE THREAD

**Issue by realgam3** [pymodsecurity/pymodsecurity#14]
# The Issue:
Using setServerLogCb Causes Python Process To Stay Open Forever

# Code Examples:
### Callback.py (Will Be Stuck After Execution)
```python
from ModSecurity import Rules
from ModSecurity import testLogCb
from ModSecurity import Transaction
from ModSecurity import LogProperty
from ModSecurity import ModSecurity


def serverlogcb(a1, a2):
    print('callback from modsec')
    print(a1, a2)


modsec = ModSecurity()
print(modsec.whoAmI())

modsec.setServerLogCb(
    serverlogcb, LogProperty.RuleMessageLogProperty)

testLogCb("a", "b") # helper function

rules = Rules()

rules.loadFromUri("basic_rules.conf")
print(rules.getParserError())

transaction = Transaction(modsec, rules)
print(transaction.processURI(
    "http://www.modsecurity.org/test?key1=value1&key2=value2&key3=value3&test=args&test=test", "GET", "2.0"))
```

### Callback-Strace.txt (Show Why It Is Stuck)
```
write(1, "1\n", 2)                      = 2
rt_sigaction(SIGINT, {sa_handler=SIG_DFL, sa_mask=[], sa_flags=SA_RESTORER, sa_restorer=0x7f8d9ea3ef20}, {sa_handler=0x630560, sa_mask=[], sa_flags=SA_RESTORER, sa_restorer=0x7f8d9ea3ef20}, 8) = 0
sigaltstack(NULL, {ss_sp=0xaf5530, ss_flags=0, ss_size=8192}) = 0
sigaltstack({ss_sp=NULL, ss_flags=SS_DISABLE, ss_size=0}, NULL) = 0
gettimeofday({tv_sec=1576144954, tv_usec=3740}, NULL) = 0
futex(0xa745c8, FUTEX_WAIT_BITSET_PRIVATE|FUTEX_CLOCK_REALTIME, 0, {tv_sec=1576144954, tv_nsec=8740000}, 0xffffffff) = -1 ETIMEDOUT (Connection timed out)
gettimeofday({tv_sec=1576144954, tv_usec=9729}, NULL) = 0
futex(0xa74560, FUTEX_WAKE_PRIVATE, 1)  = 0
futex(0xa745c8, FUTEX_WAIT_BITSET_PRIVATE|FUTEX_CLOCK_REALTIME, 0, {tv_sec=1576144954, tv_nsec=14729000}, 0xffffffff) = -1 ETIMEDOUT (Connection timed out)
gettimeofday({tv_sec=1576144954, tv_usec=15713}, NULL) = 0
futex(0xa74560, FUTEX_WAKE_PRIVATE, 1)  = 0
futex(0xa745c8, FUTEX_WAIT_BITSET_PRIVATE|FUTEX_CLOCK_REALTIME, 0, {tv_sec=1576144954, tv_nsec=20713000}, 0xffffffff) = -1 ETIMEDOUT (Connection timed out)
gettimeofday({tv_sec=1576144954, tv_usec=21702}, NULL) = 0
futex(0xa74560, FUTEX_WAKE_PRIVATE, 1)  = 0
futex(0xa745c8, FUTEX_WAIT_BITSET_PRIVATE|FUTEX_CLOCK_REALTIME, 0, {tv_sec=1576144954, tv_nsec=26702000}, 0xffffffff) = -1 ETIMEDOUT (Connection timed out)
gettimeofday({tv_sec=1576144954, tv_usec=27723}, NULL) = 0
futex(0xa74560, FUTEX_WAKE_PRIVATE, 1)  = 0
```

## Cause
This Syscalls Run On Endless Loop
```
futex(0xa745c8, FUTEX_WAIT_BITSET_PRIVATE|FUTEX_CLOCK_REALTIME, 0, {tv_sec=1576144954, tv_nsec=8740000}, 0xffffffff) = -1 ETIMEDOUT (Connection timed out)
gettimeofday({tv_sec=1576144954, tv_usec=9729}, NULL) = 0
futex(0xa74560, FUTEX_WAKE_PRIVATE, 1)  = 0
```

### No-Callback.py (Will Be Closed After Execution)
```python
from ModSecurity import Rules
from ModSecurity import Transaction
from ModSecurity import ModSecurity


modsec = ModSecurity()
print(modsec.whoAmI())

rules = Rules()

rules.loadFromUri("basic_rules.conf")
print(rules.getParserError())

transaction = Transaction(modsec, rules)
print(transaction.processURI(
    "http://www.modsecurity.org/test?key1=value1&key2=value2&key3=value3&test=args&test=test", "GET", "2.0"))
```

### No-Callback-Strace.txt (Show Proper Closing)
```
write(1, "1\n", 2)                      = 2
rt_sigaction(SIGINT, {sa_handler=SIG_DFL, sa_mask=[], sa_flags=SA_RESTORER, sa_restorer=0x7f809183ef20}, {sa_handler=0x630560, sa_mask=[], sa_flags=SA_RESTORER, sa_restorer=0x7f809183ef20}, 8) = 0
shmctl(0, IPC_RMID, NULL)               = 0
munmap(0x7f80876c0000, 266240)          = 0
sigaltstack(NULL, {ss_sp=0xcce540, ss_flags=0, ss_size=8192}) = 0
sigaltstack({ss_sp=NULL, ss_flags=SS_DISABLE, ss_size=0}, NULL) = 0
exit_group(0)                           = ?
+++ exited with 0 +++
```

# Test Environment
OS: Ubuntu 18.04.3 LTS
Kernel: 4.4.0-18362

Python 3.6.9
pymodsecurity 0.0.4

ModSecurity - v3.0.3 for Linux

#### Mandatory dependencies
   + libInjection                                  ....v3.9.2-30-gbf234eb
   + SecLang tests                                 ....e6b03e4

#### Optional dependencies
   + GeoIP/MaxMind                                 ....found
      * (GeoIP) v1.6.12
   + LibCURL                                       ....found v7.58.0
   + YAJL                                          ....found v2.1.0
   + LMDB                                          ....not found
   + LibXML2                                       ....found v2.9.4
   + SSDEEP                                        ....found
   + LUA                                           ....not found

#### Other Options
   + Test Utilities                                ....enabled
   + SecDebugLog                                   ....enabled
   + afl fuzzer                                    ....disabled
   + library examples                              ....enabled
   + Building parser                               ....disabled
   + Treating pm operations as critical section    ....disabled

**Comment by realgam3** [comment]
I found a non elegant way to solve this issue: https://github.com/realgam3/pymodsecurity/commit/ce57006e8590cd6c64a36d8ea49fb16338d15aa4
But I don't think its the right way to solve it....

**Comment by GustavoKatel** [comment]
Hi @realgam3 ! Sorry for the long delay. This has been fixed in v0.0.5. Your workaround showed me the way to the actual fix. 

I will close this and feel free to reopen if this is still happenning.
